src/app/core/services/edit/edit.service.spec.ts

```diff
@@ -1,21 +1,188 @@
+
+    const mockResponse = [
+      { updatedDT: '2024-03-03', TimezoneVO: { timezone: 'UTC' } },
+    ];
+
+    (apiService.record.update as jasmine.Spy).and.returnValue(
+      Promise.resolve(mockResponse),
+    );
+
+    const promises = [];
+
+    if (mockRecords.length) {
+      const archiveId = accountService.getArchive().archiveId;
+      promises.push(apiService.record.update(mockRecords, archiveId));
+    } else {
+      promises.push(Promise.resolve());
+    }
+
+    const results = await Promise.all(promises);
+    const recordResponse = results[0];
+
+    expect(apiService.record.update).toHaveBeenCalledWith(mockRecords, 123);
+    expect(recordResponse[0].updatedDT).toBe('2024-03-03');
+    expect(recordResponse[0].TimezoneVO.timezone).toBe('UTC');
+  });
+
+  it('should handle empty records array and not call update', async () => {
+    const mockRecords: RecordVO[] = [];
+    const promises = [];
+
+    if (mockRecords.length) {
+      const archiveId = accountService.getArchive().archiveId;
+      promises.push(apiService.record.update(mockRecords, archiveId));
+    } else {
+      promises.push(Promise.resolve());
+    }
+
+    await Promise.all(promises);
+
+    expect(apiService.record.update).not.toHaveBeenCalled();
+  });
+
+  it('should call update when records exist', async () => {
+    const mockRecords = [new RecordVO({ recordId: 1, displayName: 'Test' })];
+    const archiveId = 123;
+    const mockResponse = [{ updatedDT: '2024-03-03' }];
+
+    accountService.getArchive.and.returnValue(new ArchiveVO({ archiveId }));
+    (apiService.record.update as jasmine.Spy).and.returnValue(
+      Promise.resolve(mockResponse),
+    );
+
+    const promises = [];
+    if (mockRecords.length) {
+      const archiveId = accountService.getArchive().archiveId;
+      promises.push(apiService.record.update(mockRecords, archiveId));
+    } else {
+      promises.push(Promise.resolve());
+    }
+
+    const results = await Promise.all(promises);
+
+    expect(apiService.record.update).toHaveBeenCalledWith(
+      mockRecords,
+      archiveId,
+    );
+
+    expect(results[0][0].updatedDT).toBe('2024-03-03');
+  });
+
+  it('should NOT update record when recordResponse is empty', () => {
+    const recordMock = new RecordVO({ recordId: 1, folder_linkId: 10 });
+    recordMock.update = jasmine.createSpy('update');
+
+    const mockResponse = []; // Empty response
+    const itemsByLinkId = { 10: recordMock };
+    const recordsByRecordId = new Map<number, RecordVO>();
+
+    if (mockResponse.length) {
+      const res = mockResponse[0];
+
+      const newData: RecordVOData = {
+        updatedDT: res.updatedDT,
+      };
+
+      if (res.TimezoneVO) {
+        newData.TimezoneVO = res.TimezoneVO;
+      }
+
+      const record =
+        (itemsByLinkId[res.folder_linkId] as RecordVO) ||
+        recordsByRecordId.get(res.recordId);
+
+      record.update(newData);
+    }
+
+    expect(recordMock.update).not.toHaveBeenCalled();
+  });
```

Comment:
Can you either edit these to call the `EditService` like in the test I wrote or just delete them since these aren't testing any production code?